test/jdk/java/awt/Frame/InvisibleOwner/InvisibleOwner.java

```diff
@@ -24,99 +24,159 @@
+        ownedWindow = new Window(invisibleFrame);
+        ownedWindow.setBackground(Color.RED);
+        ownedWindow.setLocation(H_X+H_W+100, H_Y+H_W+100);
+        ownedWindow.setSize(100, 100);
+        ownedWindow.setVisible(true);
 
-        Robot robot = new Robot();
+        Toolkit.getDefaultToolkit().sync();
+    }
 
-        // Clicking the owned window shouldn't make its owner visible
-        Util.clickOnComp(window, robot);
-        try { Thread.sleep(500); } catch (Exception ex) {}
+    static void captureScreen() throws Exception {
+        System.out.println("Writing screen capture");
+        Rectangle screenRect = helperFrame.getGraphicsConfiguration().getBounds();
+        java.awt.image.BufferedImage bi = robot.createScreenCapture(screenRect);
+        javax.imageio.ImageIO.write(bi, "png", new java.io.File("screen_IO.png"));
+    }
 
+    public static void main(String[] args) throws Exception {
 
-        // Assume the location and size are applied to the frame as expected.
-        // This should work fine on the Mac. We can't call getLocationOnScreen()
-        // since from Java perspective the frame is invisible anyway.
+        try {
+            EventQueue.invokeAndWait(() -> createUI());
+            robot = new Robot();
+            robot.waitForIdle();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
 
-        // 1. Check the color at the center of the owner frame
-        Color c = robot.getPixelColor(F_X + F_W / 2, F_Y + F_H / 2);
-        System.err.println("Pixel color: " + c);
-        if (c == null) {
-            throw new RuntimeException("Robot.getPixelColor() failed");
-        }
-        if (c.equals(frame.getBackground())) {
-            throw new RuntimeException("The invisible frame has become visible");
-        }
-        if (!c.equals(helperFrame.getBackground())) {
-            throw new RuntimeException("The background helper frame has been covered by something unexpected");
-        }
+            Rectangle helperBounds = helperFrame.getBounds();
+            System.out.println("helperFrame bounds = " + helperBounds);
+            if (!helperBounds.contains(C_X, C_Y)) {
+                System.out.println("Helper not positioned where it needs to be");
+                return;
+            }
 
-        // 2. Try to click it
-        robot.mouseMove(F_X + F_W / 2, F_Y + F_H / 2);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        try { Thread.sleep(500); } catch (Exception ex) {}
+            // Clicking the owned window shouldn't make its owner visible
+            Rectangle ownedWindowBounds = ownedWindow.getBounds();
+            robot.mouseMove(ownedWindowBounds.x + ownedWindowBounds.width / 2,
+                            ownedWindowBounds.y + ownedWindowBounds.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_MASK);
```

Comment:
Nit: Shouldn't we use BUTTON1_DOWN_MASK instead of deprecated BUTTON1_MASK?

Reply:
Thought that we are cleaning up such deprecated things.
There is another place with BUTTON1_MASK below though.